docs/defining_new_slo.md

```diff
@@ -0,0 +1,191 @@
+This can be easy to implement, and you can find number of upstream
+projects that can help you with that such as [this SLO generator](https://promtools.dev/alerts/errors), 
+[Sloth](https://sloth.dev/) or [the Pyrra project](https://github.com/pyrra-dev/pyrra) having even a nice visualization.
+
+#### Advantages
+- No need for application changes, you can use already existing Prometheus metrics you already collect
+- No need for other infrastructure changes
+
+#### Disadvantages
+- Limited by cardinality because of the Prometheus architecture and purpose
+  (if needed for filtering of events or having more precise thresholds for some events)
+- For latency and quality SLI types need to use histograms and its buckets need to match the thresholds
+- Computing SLO based on high cardinality metrics over 4w can be enormously demanding for the Prometheus cluster
+
+## Using slo-exporter
+Slo exporter was built to solve more complex SLO setups and to unify SLO computation.
+It has support to compute SLO based on querying Prometheus data, tailing logs from file, receiving logs from Envoy proxy, following Kafka topic or reading logs from Elasticsearch.
+
+### 1. Choose source of the events
+#### Querying Prometheus
+This mode is mostly just for unification of the SLO computation so everyone uses the same tooling.
+It could be replaced by Prometheus recording rules and just hooking into the alerting by using the same metric naming schema as slo-exporter.
+It also does not support HA setup, so it's not advised to use this and avoid it if possible.
+Slo-exporter documentation can be found [here](https://github.com/seznam/slo-exporter/blob/master/docs/modules/prometheus_ingester.md) and complete example with whole configuration and kubernetes manifests
+[here](https://github.com/seznam/slo-exporter/tree/master/examples/prometheus).
+
+> Warning: in this setup, the slo-exporter cannot run in multiple replicas because it would lead to duplicating the observed events. Make sure to alert on that.
+
+#### From Envoy access logs
+If your service is accessed by users via Envoy proxy, you can configure it to send the access logs using gRPC to the slo-exporter.
+Documentation about the slo-exporter gRPC logging module is [here](https://github.com/seznam/slo-exporter/blob/master/docs/modules/envoy_access_log_server.md),
+documentation how to configure Envoy to send the access logs over gRPC can be found [here](https://www.envoyproxy.io/docs/envoy/latest/api-v3/data/accesslog/v3/accesslog.proto).
+Complete example of the setup is [here](https://github.com/seznam/slo-exporter/tree/master/examples/envoy_proxy).
+
+#### Elasticsearch
+> Currently in experimental mode
+
+In case you want to compute the SLO based on your application logs and use an ELK stack, you can hook the slo-exporter to it.
+
+> Warning: in this setup, the slo-exporter cannot run in multiple replicas because it would lead to duplicating the observed events. Make sure to be alert on that.
```

Comment:
This should be primarily in the elasticsearch module's documentation, which I now see is completely missing :thinking:

Reply:
Because it's not in the master :P

Reply:
I'll drop the whole ELK part

Reply:
https://github.com/seznam/slo-exporter/pull/86/commits/8124d42eef4dadd3ca48e00ae52980040e02e07f